urlc8dbc7ce6e_dlubarov_guava_tgz-pJ8-MainJ8/src/parse/MethodDefParser.java

```diff
@@ -0,0 +1,183 @@
+package parse;
+
+import java.util.*;
+
+import common.NiftyException;
+
+import parse.misc.*;
+import parse.stm.BlockParser;
+import a.*;
+import a.gen.GenericConstraint;
+import a.stm.Block;
+
+public class MethodDefParser extends Parser<MemberDef> {
+    public static final Parser<MemberDef> singleton = new MethodDefParser();
+    private MethodDefParser() {}
+
+    private static String[] methodQualifiers = {"public", "private", "static", "sealed"};
+
+    private static boolean isMethodQualifier(String s) {
+        for (String qual : methodQualifiers)
+            if (s.equals(qual))
+                return true;
+        return false;
+    }
+
+    private static Success<String> parseMethodQualifier(String s, int p) {
+        Success<String> resQual = IdentifierParser.singleton.parse(s, p);
+        if (resQual != null && isMethodQualifier(resQual.value))
+            return resQual;
+        return null;
+    }
+
+    @Override
+    public Success<MemberDef> parse(String s, int p) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.